server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestNodesInfoAction.java

```diff
@@ -108,16 +98,9 @@ static NodesInfoRequest prepareRequest(final RestRequest request) {
             nodesInfoRequest.all();
         } else {
             nodesInfoRequest.clear();
-            nodesInfoRequest.settings(metrics.contains("settings"));
-            nodesInfoRequest.os(metrics.contains("os"));
-            nodesInfoRequest.process(metrics.contains("process"));
-            nodesInfoRequest.jvm(metrics.contains("jvm"));
-            nodesInfoRequest.threadPool(metrics.contains("thread_pool"));
-            nodesInfoRequest.transport(metrics.contains("transport"));
-            nodesInfoRequest.http(metrics.contains("http"));
-            nodesInfoRequest.plugins(metrics.contains("plugins"));
-            nodesInfoRequest.ingest(metrics.contains("ingest"));
-            nodesInfoRequest.indices(metrics.contains("indices"));
+            // disregard unknown metrics
```

Comment:
Aren't unknown metrics an error case?

Reply:
The current behavior of the Nodes Info API is to silently disregard invalid metrics (or node ids). The Nodes Stats endpoint, on the other hand, will return an error if you request an unknown metric.

Arguably, the Nodes Info API _should_ return an error for unknown metrics, but that would go beyond just refactoring.